app/v2/services/XmlParsers.scala

```diff
@@ -20,24 +20,30 @@ import akka.NotUsed
 import akka.stream.alpakka.xml.ParseEvent
 import akka.stream.alpakka.xml.StartElement
 import akka.stream.scaladsl.Flow
+import v2.models.MovementType
 import v2.models.errors.ExtractionError
 import v2.models.request.MessageType
 
 object XmlParsers {
 
-  val messageTypeExtractor: Flow[ParseEvent, Either[ExtractionError, MessageType], NotUsed] = Flow[ParseEvent]
+  def messageTypeExtractor(movementType: MovementType): Flow[ParseEvent, Either[ExtractionError, MessageType], NotUsed] = Flow[ParseEvent]
     .filter {
       case _: StartElement =>
         true
       case _ => false
     }
     .take(1)
     .map {
-      case s: StartElement =>
+      case s: StartElement if movementType == MovementType.Departure =>
         MessageType.messageTypesSentByDepartureTrader
           .find(_.rootNode == s.localName)
           .map(Right(_))
           .getOrElse(Left(ExtractionError.MessageTypeNotFound(s.localName)))
+      case s: StartElement if movementType == MovementType.Arrival =>
+        MessageType.messageTypesSentByArrivalTrader
+          .find(_.rootNode == s.localName)
```

Comment:
A little nitpicky, but there's a bit of duplication here that I'd like to see removed, just for ease of maintenance. Something like:

```scala
sentTypesFor(MovementType)
   .find(find(_.rootNode == s.localName) // etc...
```

or perhaps creating a link to the sent types in the `MovementType` object itself.

Reply:
done the changes as suggested